src/webui/utils/login.js

```diff
@@ -0,0 +1,78 @@
+import isString from 'lodash/isString';
+import isNumber from 'lodash/isNumber';
+import size from 'lodash/size';
+import {Base64} from 'js-base64';
+import API from './api';
+import {HEADERS} from '../../lib/constants';
+
+export function isTokenExpire(token) {
+    if (!isString(token)) {
+        return true;
+    }
+
+    let payload = token.split('.')[1];
+
+    if (!payload) {
+        return true;
+    }
+
+    try {
+        payload = JSON.parse(Base64.decode(payload));
+    } catch (err) {
+        console.error('Invalid token:', err, token); // eslint-disable-line
+        return false;
+    }
+
+    if (!payload.exp || !isNumber(payload.exp)) {
+        return true;
+    }
+    // Report as expire before (real expire time - 30s)
+    const jsTimestamp = payload.exp * 1000 - 30000;
```

Comment:
I'm not 100% sure what you are doing here. 🤔

Reply:
It's from the start, Even I could not figure out this.

Reply:
Oh,  The Horror! 😆

Reply:
What unit is `payload.exp` in? Milliseconds or seconds?
Could be a little clearer with:
```js
(payload.exp - 30) * 1000; // seconds
```
or
```js
payload.exp - (30 * 1000) // milliseconds
```